vassal-app/src/main/java/VASSAL/build/module/map/DrawPile.java

```diff
@@ -82,6 +82,19 @@
 
   private final VisibilityCondition restrictExpressionVisibleCondition = () -> dummy.isRestrictOption();
 
+  private final VisibilityCondition drawMultipleMessageVisibleCondition = () -> dummy.isAllowMultipleDraw();
+
+  private final VisibilityCondition drawSpecificMessageVisibleCondition = () -> dummy.isAllowSelectDraw();
+
+  private final VisibilityCondition faceUpDownMessageVisibleCondition = () -> Deck.USE_MENU.equals(dummy.getFaceDownOption());
+
+  private final VisibilityCondition sortVisibleCondition = () -> dummy.isSortable();
+
+  private final VisibilityCondition selectOrSortVisibleCondition = () -> dummy.isSortable() || dummy.isAllowSelectDraw();
+
```

Comment:
Is using the same property name for Drawing and Sorting flexible enough if you want to do both? Sorting for drawing will usually by by some sensible 'user meaningful' order, but sorting for game play purposes will quite likely be completely different.

What about if the designer wants two kinds of sorts for different times during the game? Perhaps the Sort Command should be taken out of the DrawPile itself and added as a Child like Deck Global Key Commands are? That will also simplify the rapidly expanding DrawPile config screen. 

I would leave the existing sort property and the new descending? option for card draw display. Add a separate Sort property and descending flag to the new child Sort command.

What about multi-level sorts? Tribe by descending Strength? <evil grin> This can be handled by creating a Calculated Property with the compound sort key so probably not needed.

Reply:
In fact, the 'Command to send entire deck to another deck' should probably be a child as well as I can easily see the need for multiple of these. For this, maybe just leave the one that is in the Deck there, but add the facility to add extras as children. This would keep compatibility without a painful version conversion.